### PR TITLE
Sp cleanup

### PR DESCRIPTION
**Pull request by nirs**
Various cleanups

---
This change is [Reviewable](https://reviewable.io/reviews/ovirt/vdsm/5)

### CODE REVIEW

File: vdsm/storage/blockSD.py [hunk 1]
```diff
@@ -862,15 +862,13 @@ def _registerResourceNamespaces(self):
...
-        rmanager = rm.ResourceManager.getInstance()
         # Register lvm activation resource namespace for the underlying VG
         lvmActivationFactory = resourceFactories.LvmActivationFactory(
             self.sdUUID)
         lvmActivationNamespace = sd.getNamespace(sc.LVM_ACTIVATION_NAMESPACE,
                                                  self.sdUUID)
         try:
-            rmanager.registerNamespace(lvmActivationNamespace,
-                                       lvmActivationFactory)
+            rm.registerNamespace(lvmActivationNamespace, lvmActivationFactory)
```
Comment (aglitke) [inline vdsm/storage/blockSD.py]: Nice!  More concise.

File: vdsm/storage/resourceFactories.py [hunk 2]
```diff
@@ -152,20 +150,20 @@ def __getResourceCandidatesList(self, resourceName, lockType):
...
                         self.volumeResourcesNamespace,
                         template, lockType,
                         timeout=self.resource_default_timeout)
                 volResourcesList.append(volRes)
 
             # Acquire 'lockType' volume locks
             for volUUID in volUUIDChain:
-                volRes = rmanager.acquireResource(
+                volRes = rm.acquireResource(
                     self.volumeResourcesNamespace,
```
Comment (aglitke) [inline vdsm/storage/resourceFactories.py]: Looks good

File: vdsm/storage/resourceManager.py [hunk 3]
```diff
@@ -336,31 +334,17 @@ class ResourceManager(object):
...
-        """
-        Get the global resource manager
-        """
-        if cls._instance is None:
-            with cls._singletonLock:
-                if cls._instance is None:
-                    cls._instance = ResourceManager()
-
-        return cls._instance
-
```
Comment (aglitke) [inline vdsm/storage/resourceManager.py]: Prayer answered!  Great!

File: tests/storage_resourcemanager_test.py [hunk 4]
```diff
@@ -697,19 +691,18 @@ def test_mode_ignored_for_sorting(self):
         self.assertFalse(a < b)
         self.assertFalse(b < a)
 
+    @MonkeyPatch(rm, "_manager", FakeResourceManager())
```
Comment (aglitke) [inline tests/storage_resourcemanager_test.py]: Nice!

File: vdsm/storage/resourceManager.py [hunk 5]
```diff
@@ -330,7 +330,7 @@ def __repr__(self):
             repr(self.__wrappedObject) if self.isValid else None)
 
 
-class ResourceManager(object):
+class _ResourceManager(object):
```
Comment (aglitke) [inline vdsm/storage/resourceManager.py]: ok.

File: vdsm/storage/sp.py [hunk 6]
```diff
@@ -1342,9 +1342,8 @@ def extendVolume(self, sdUUID, volumeUUID, size, isShuttingDown=None):
         sdCache.produce(sdUUID).extendVolume(volumeUUID, size, isShuttingDown)
 
     def extendVolumeSize(self, sdUUID, imgUUID, volUUID, newSize):
-        imageResourcesNamespace = sd.getNamespace(sc.IMAGE_NAMESPACE, sdUUID)
-        with rm.acquireResource(imageResourcesNamespace, imgUUID,
-                                rm.EXCLUSIVE):
+        img_ns = sd.getNamespace(sc.IMAGE_NAMESPACE, sdUUID)
+        with rm.acquireResource(img_ns, imgUUID, rm.EXCLUSIVE):
```
Comment (aglitke) [inline vdsm/storage/sp.py]: A no brainer.  Great!

File: vdsm/storage/sp.py [hunk 7]
```diff
@@ -1523,18 +1523,14 @@ def copyImage(self, sdUUID, vmUUID, srcImgUUID, srcVolUUID, dstImgUUID,
...
         else:
-            dstImageResourcesNamespace = srcImageResourcesNamespace
+            dst_img_ns = src_img_ns
 
-        with nested(rm.acquireResource(srcImageResourcesNamespace,
-                                       srcImgUUID, rm.SHARED),
-                    rm.acquireResource(dstImageResourcesNamespace,
-                                       dstImgUUID, rm.EXCLUSIVE)
+        with nested(rm.acquireResource(src_img_ns, srcImgUUID, rm.SHARED),
+                    rm.acquireResource(dst_img_ns, dstImgUUID, rm.EXCLUSIVE)
```
Comment (aglitke) [inline vdsm/storage/sp.py]: :)

**Comment by nirs** [comment]
For the resourceManager cleanups, better review #6.

File: vdsm/storage/sp.py [hunk 8]
```diff
@@ -1678,10 +1678,8 @@ def uploadImage(self, methodArgs, sdUUID, imgUUID, volUUID=None):
         Upload an image to a remote endpoint using the specified method and
         methodArgs.
         """
-        imgResourceLock = rm.acquireResource(
-            sd.getNamespace(sc.IMAGE_NAMESPACE, sdUUID), imgUUID, rm.SHARED)
-
-        with imgResourceLock:
+        img_ns = sd.getNamespace(sc.IMAGE_NAMESPACE, sdUUID)
+        with rm.acquireResource(img_ns, imgUUID, rm.SHARED):
```
Comment (aglitke) [inline vdsm/storage/sp.py]: Great.

File: vdsm/storage/sp.py [hunk 9]
```diff
@@ -426,13 +426,10 @@ def _upgradePool(self, targetDomVersion, lockTimeout=None):
...
-                self._domainsToUpgrade = self.getDomains(activeOnly=True)\
-                    .keys()
-                try:
-                    self._domainsToUpgrade.remove(self.masterDomain.sdUUID)
-                except ValueError:
-                    pass
+                self.log.debug("Marking active domains for upgrade")
+                domains = self.getDomains(activeOnly=True)
+                domains.pop(self.masterDomain.sdUUID, None)
+                self._domainsToUpgrade = domains.keys()
```
Comment (aglitke) [inline vdsm/storage/sp.py]: Nice!

File: vdsm/storage/sp.py [hunk 10]
```diff
@@ -1340,8 +1340,8 @@ def extendVolume(self, sdUUID, volumeUUID, size, isShuttingDown=None):
     def extendVolumeSize(self, sdUUID, imgUUID, volUUID, newSize):
         img_ns = sd.getNamespace(sc.IMAGE_NAMESPACE, sdUUID)
         with rm.acquireResource(img_ns, imgUUID, rm.EXCLUSIVE):
-            return sdCache.produce(sdUUID) \
-                .produceVolume(imgUUID, volUUID).extendSize(int(newSize))
+            vol = sdCache.produce(sdUUID).produceVolume(imgUUID, volUUID)
+            return vol.extendSize(int(newSize))
```
Comment (aglitke) [inline vdsm/storage/sp.py]: Much better.

File: vdsm/storage/sp.py [hunk 11]
```diff
@@ -1848,8 +1848,7 @@ def createVolume(self, sdUUID, imgUUID, size, volFormat, preallocate,
```
Comment (aglitke) [inline vdsm/storage/sp.py]: procudeVolume is misspelled in the commit message.

File: vdsm/storage/sp.py [hunk 12]
```diff
@@ -1848,8 +1848,7 @@ def createVolume(self, sdUUID, imgUUID, size, volFormat, preallocate,
 
         if imgUUID != srcImgUUID and srcImgUUID != sc.BLANK_UUID:
             srcDom = sdCache.produce(sdUUID)
-            srcVol = srcDom.produceVolume(imgUUID=srcImgUUID,
-                                          volUUID=srcVolUUID)
+            srcVol = srcDom.produceVolume(srcImgUUID, srcVolUUID)
```
Comment (aglitke) [inline vdsm/storage/sp.py]: good.

File: vdsm/storage/sp.py [hunk 13]
```diff
@@ -1891,8 +1891,8 @@ def deleteVolume(self, sdUUID, imgUUID, volumes, postZero, force):
         with rm.acquireResource(img_ns, imgUUID, rm.EXCLUSIVE):
             dom = sdCache.produce(sdUUID)
             for volUUID in volumes:
-                dom.produceVolume(imgUUID, volUUID).delete(
-                    postZero=postZero, force=force)
+                vol = dom.produceVolume(imgUUID, volUUID)
+                vol.delete(postZero=postZero, force=force)
```
Comment (aglitke) [inline vdsm/storage/sp.py]: Much easier to read.

File: vdsm/storage/sp.py [hunk 14]
```diff
@@ -1942,14 +1942,14 @@ def setVolumeDescription(self, sdUUID, imgUUID, volUUID, description):
         img_ns = sd.getNamespace(sc.IMAGE_NAMESPACE, sdUUID)
         with rm.acquireResource(img_ns, imgUUID, rm.EXCLUSIVE):
             vol = sdCache.produce(sdUUID).produceVolume(imgUUID, volUUID)
-            vol.setDescription(descr=description)
+            vol.setDescription(description)
```
Comment (aglitke) [inline vdsm/storage/sp.py]: ok

File: vdsm/storage/sp.py [hunk 15]
```diff
@@ -1528,7 +1528,8 @@ def copyImage(self, sdUUID, vmUUID, srcImgUUID, srcVolUUID, dstImgUUID,
         with nested(rm.acquireResource(src_img_ns, srcImgUUID, rm.SHARED),
                     rm.acquireResource(dst_img_ns, dstImgUUID, rm.EXCLUSIVE)
                     ):
-            dstUUID = image.Image(self.poolPath).copyCollapsed(
+            img = image.Image(self.poolPath)
+            dstUUID = img.copyCollapsed(
```
Comment (aglitke) [inline vdsm/storage/sp.py]: better.

File: vdsm/storage/sp.py [hunk 16]
```diff
@@ -316,8 +316,8 @@ def startSpm(self, prevID, prevLVER, maxHostID, expectedDomVersion=None):
                 # mailbox and the "self._setSecure() call"
 
                 # FIXME : Use a system wide grouping mechanism
-                if (self.lvExtendPolicy == "ON"
-                        and self.masterDomain.supportsMailbox):
+                if (self.lvExtendPolicy == "ON" and
+                        self.masterDomain.supportsMailbox):
```
Comment (aglitke) [inline vdsm/storage/sp.py]: ok.

File: vdsm/storage/sp.py [hunk 17]
```diff
@@ -1367,7 +1367,6 @@ def getInfo(self):
             'master_ver': self._backend.getMasterVersion(),
             'lver': LVER_INVALID,
             'spm_id': SPM_ID_FREE,
-            'pool_status': 'uninitialized',
```
Comment (aglitke) [inline vdsm/storage/sp.py]: ok.

File: vdsm/storage/sp.py [hunk 18]
```diff
@@ -1360,16 +1360,16 @@ def getInfo(self):
...
             'master_uuid': self.masterDomain.sdUUID,
             'master_ver': self._backend.getMasterVersion(),
-            'lver': LVER_INVALID,
+            'name': '',
+            'pool_status': 'connected',
             'spm_id': SPM_ID_FREE,
+            'type': msdInfo['type'],
             'version': str(msdInfo['version']),
-            'isoprefix': '',
-            'pool_status': 'connected',
```
Comment (aglitke) [inline vdsm/storage/sp.py]: ok

File: vdsm/storage/sp.py [hunk 19]
```diff
@@ -68,7 +68,6 @@ class StoragePool(object):
 
     log = logging.getLogger('storage.StoragePool')
     storage_repository = config.get('irs', 'repository')
-    lvExtendPolicy = config.get('irs', 'vol_extend_policy')
```
Comment (aglitke) [inline vdsm/storage/sp.py]: great.

**Comment by pkliczewski** [comment]
Have a missed discussion about changing our dev process? Why are we using gh instread of gerrit?

**Comment by nirs** [comment]
@pkliczewski we are experimenting, if this works well we can discuss changing the process.

**Comment by nirs** [comment]
@aglitke thanks for reviewing this, the current version fixes few typos in the commit messages.

**Comment by nirs** [comment]
Merged.

**Pull request closed without merging**